### ISSUE TITLE
Can you replace lmdbjava library in dsitributed data with something else

### ISSUE THREAD

**Issue by mehmetsalgar** [akka/akka#31130]
Hi,

this library has problems with JDK17 and I don't see anything done about it.

#30341 

Today I also discovered I can't develop with Akka with my new Mac Pro with M1 chip because of lmdbjava....

They have a one year old issue, [M1 support lmdbjava](https://github.com/lmdbjava/native/issues/10) and no action, I think you can't rely a library like this.

**Comment by patriknw** [comment]
lmdb is only used when the durable feature of Distributed Data or remember entities feature of Cluster Sharding (with ddata store) has been enabled. Those are not enabled by default. Is that what you use or do you a problem also when not using these features?

**Comment by mehmetsalgar** [comment]
Yes, unfortunately I need the rememberEntities functionality and durable future.

**Comment by johanandren** [comment]
Running a x86_64 JDK with rosetta on a M1 does work as an alternative to contributing an upstream PR to lmdbjava adding aarch64 support.

If you want to run on JDK 17 you also need to open up nio for reflective access with the java argument `--add-opens=java.base/java.nio=ALL-UNNAMED`, like we do in our nightly JDK 17 tests of Akka. (We have a still undone task of adding some details about that to the docs #31087) For multi node tests and such where it can be hard to pass arbitrary parameter you may need to set it using something like the `JAVA_TOOL_OPTS` env variable.

**Comment by mehmetsalgar** [comment]
For JDK and relaxing security, when my company completes JDK17 migration, I don't think, I can convince them to open security gates that JDK developer think that it should stay closed.

But as I said, currently that is a secondary concern, my more urgent problem making Akka Cluster Sharding <-> ddata working with my M1 Mac. I have Rosetta installed, let's see how that works.

**Comment by patriknw** [comment]
For rememberEntities I would recommend using the eventsourced mode for remember entities, see https://doc.akka.io/docs/akka/current/typed/cluster-sharding.html#event-sourced-mode

Then would use Akka Persistence, and your choice of database plugin, for storage of this.

**Comment by johanandren** [comment]
Looks like lmdbjava has added cross builds, but there is no release including it yet: https://github.com/lmdbjava/lmdbjava/pull/218

**Comment by johanandren** [comment]
Just tried it and 0.9.0 works on apple silicon, however there are concerns about transitive dependency problems that might block us from upgrading (see https://github.com/akka/akka/pull/29306#issuecomment-651049733 )
